lib/semian.rb

```diff
@@ -216,7 +217,7 @@ def unregister_all_resources
 
   # Retrieves a hash of all registered resources.
   def resources
-    @resources ||= {}
+    @resources ||= LRUHash.new(minimum_time_in_lru: 300)
```

Comment:
If we're gonna hardcode 300, I'd rather we don't hardcode it twice.
A constant `MINIMUM_TIME_IN_LRU` or a `new_lru_hash` method would be my preference.